File: design-patterns/ts/visitor/ui-element/wrap-element.ts
// wrap-element.ts

import UIElement from "./ui-element";
import Visitor from "./visitor";

class WrapElement implements UIElement {
    constructor(private text: string, private wrapper: string) {

    }

    getText(): string {
        return this.text
    }

    getWrapper(): string {
        return this.wrapper;
    }

    appendElement(visitor: Visitor): void {
        visitor.appendContentWithWrapper(this);
    }
}

export default WrapElement;

File: design-patterns/go/visitor/ui_element/wrap_element.go
// wrap_element.go

package main

type WrapElement struct {
	text    string
	wrapper string
}

func NewWrapElement(text string, wrapper string) (wrapElement *WrapElement) {
	wrapElement = &WrapElement{}
	wrapElement.text = text
	wrapElement.wrapper = wrapper
	return
}

func (wrapElement *WrapElement) GetText() string {
	return wrapElement.text
}

func (wrapElement *WrapElement) GetWrapper() string {
	return wrapElement.wrapper
}

func (wrapElement *WrapElement) AppendElement(visitor Visitor) {
	visitor.AppendContentWithWrapper(wrapElement)
}
